### ISSUE TITLE
Host on F-Droid

### ISSUE THREAD

**Issue by tepf** [serso/android-calculatorpp#166]
I really like that you didn't dumb this calculator down like most do while maintaining a good usability. I am sure the free software loving people on [F-Droid](https://f-droid.org/) share this opinion with me.

Hopefully you consider offering it in their market as well.

**Comment by jounathaen** [comment]
+1 This App would be a worthy addition to the Repositories

**Comment by MarcRdC** [comment]
I just filed a [request](https://gitlab.com/fdroid/rfp/issues/563) for this app to be included in F‐Droid repository.
@serso Are you O.K. with that?

**Comment by IzzySoft** [comment]
Unfortunately, this app does not comply to the inclusion criteria. Is there a chance for a FOSS build-flavor that comes without the JARs, without GMS, GAds, GA, @serso?

**Comment by IzzySoft** [comment]
@serso ?? Did you abandon this app?

**Comment by MarcRdC** [comment]
I just sent @serso an e‐mail describing the situation and our little request; hope he will consider offering his help.

**Comment by serso** [comment]
Hi all, the app development is on hold now: there are no critical bugs and I don't have time/will to add new features. Regarding the FOSS build: if you want to fix it yourself that would be nice, otherwise I don't think I'll make such a build in the near future.

**Comment by IzzySoft** [comment]
Thanks, @serso – that most likely will mean "no" then, unfortunately. I'm not a dev, so I cannot work on it (and the team at F-Droid is chronically overloaded) 😢

**Comment by Bubu** [comment]
@serso I looked into this a bit, I couldn't find the source of the `org.solovyev.android.plotter` package anywhere. Did I miss anything? Is this not released as free software, by accident or on purpose?

If that was just an oversight, even if you don't do anything else with the project at the moment, could you publish this library somewhere?

**Comment by Alex-Esko** [comment]
Is this the source?
[](http://www.java2s.com/Open-Source/Android_Free_Code/Graphics/graph/serso_android_plotter.htm)

**Comment by Bubu** [comment]
This is the java source but missing at least the android xml resources. 

It should live here  https://github.com/serso/android-plotter but that's a 404

**Comment by serso** [comment]
I've uploaded the plotter library to github: https://github.com/serso/android-plotter

**Comment by Bubu** [comment]
Added to F-Droid here: https://gitlab.com/fdroid/fdroiddata/-/commit/3f71ef22bb3d3977c49bc2b80c74b72586967f6c

Built from my fork which removes Google play dependencies.

@serso Feel free to suggest any changes to the metadata file (like contact adress or donation links.)

**Comment by stevenleeS0ht** [comment]
@Bubu , Since @serso has no plan on further improvement and FOSS works on this project. Do you mind to open the ISSUE entry on the github repostory of [your fork](https://github.com/Bubu/android-calculatorpp).

Further discussion can take place under your repo.

**Comment by Bubu** [comment]
@stevenleeS0ht I'm unsure if that would make the situation any better really. I have no time or intention to take over maintenance of the project.

Someone else is free to continue on from my fork of course.

**Comment by Bubu** [comment]
FWIW: I'm moving the canonical home of the f-droid version from https://github.com/Bubu/android-calculatorpp to https://git.bubu1.eu/Bubu/android-calculatorpp
There will be a re-release as well to remove a leftover (but inert) google analytics file, I'll update the f-droid metadata then.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
@Bubu, have you moved it to continue development/maintenance?

**Comment by Bubu** [comment]
@BEEDELLROKEJULIANLOCKHART Not in any meaningful fashion I'm afraid. (beyond keeping it buildable for F-Droid)

Though If anyone would want to contribute I'd be happy to merge contributions as time permits.

**Comment by neekless** [comment]
@Bubu as of September 2023 this project has been revived. Will you update the f-droid version?

**Comment by IzzySoft** [comment]
Unlikely, as there are [some show-stoppers](https://github.com/serso/android-calculatorpp/blob/master/app/build.gradle#L27):

```gradle
apply plugin: 'com.google.gms.google-services'
apply plugin: 'com.google.firebase.crashlytics'
```

which were added with [this commit](https://github.com/serso/android-calculatorpp/commit/0a69050e98e5991f2dfb2d14f34061b1ea1ae37f).

**Comment by neekless** [comment]
> Unlikely, as there are [some show-stoppers](https://github.com/serso/android-calculatorpp/blob/master/app/build.gradle#L27):
> 
> ```groovy-gradle
> apply plugin: 'com.google.gms.google-services'
> apply plugin: 'com.google.firebase.crashlytics'
> ```
> 
> which were added with [this commit](https://github.com/serso/android-calculatorpp/commit/0a69050e98e5991f2dfb2d14f34061b1ea1ae37f).

Yes, but the point of @Bubu's fork is that Google dependencies are removed.

**Comment by IzzySoft** [comment]
Ah! OK, I didn't realize Bubu was maintaining a fork (my fault, I should have scrolled up further). Then I leave it to him to answer.

**Comment by Alex-Esko** [comment]
@serso Perhaps you want to maintain the [F-Droid](https://f-droid.org/en/packages/org.solovyev.android.calculator/) build and add a option to donate via PayPal or Liberapay.
I would be happy to make the first donation.

**Comment by Bubu** [comment]
I merged the changes from September into https://git.bubu1.eu/Bubu/android-calculatorpp and made the build easier to work with while at it.

Here's a testing version installable alongside the release version from f-droid: https://cloud.bubu1.eu/s/GAs7B5qiNPSnf76/download

We can add it to f-droid then, if this works for people.

**Comment by MarcRdC** [comment]
@Bubu Thank you. The debug version seems to work fine but has a “proxy billing activity” in it — not sure if it’s problematic or not.

**Comment by Bubu** [comment]
@MarcRdC This comes from the [checkout](https://github.com/serso/android-checkout) library I assume. (Where exactly are you seeing a proxy billing activity though?)

I hoped this would just be functional in the f-droid builds, but I tested it now and Google Play responds with "This version of the application is not configured for billing through Google Play", which I suppose is not something we can fix, as f-droid signs with a different certificate.

This basically leaves @Alex-Esko's paypal/liberapay suggestion but that depends on @serso setting this up.

**Comment by MarcRdC** [comment]
@Bubu I saw the activity when inspecting the app in [App Manager](https://f-droid.org/en/packages/io.github.muntashirakon.AppManager/).

**Comment by RokeJulianLockhart** [comment]
> as f-droid signs with a different certificate.

@Bubu, are you aware that reproducible builds would allow the same certificate? To my knowledge, that is true.

**Comment by yurtpage** [comment]
@serso if you don't have a time, you may give write permissions for the repo to someone else e.g. the @Bubu because he maintains the [F-Droid version](https://git.bubu1.eu/Bubu/android-calculatorpp). He'll review and merge PRs.
I wish to send a PR but it looks like it won't be merged: you don't have a time but Marcus don't want to make any changes except of pure FOSS version.

**Comment by Alex-Esko** [comment]
@yurtpage Where can we have a look in your PR?
What do you want to improve specifically?

**Comment by serso** [comment]
What exactly has to be done in order to make the app F-Droid compatible? Remove Firebase/Google services such as Crashlytics, Analytics and Play Store Billing? How do you build the app for F-Droid? Manually?

**Comment by yurtpage** [comment]
@Bubu can you answer the questions or send a PR?
Here are commits with of the F-Droid version
https://git.bubu1.eu/Bubu/android-calculatorpp/-/commits/main

> How do you build the app for F-Droid? Manually?

The F-Droid builds itself on CI. If the build is repdroducible then the resulted app checked to be same as the author published and the author's version is published. This allows to keep the original apk signature. Then a user can update the app from PlayStore ether from F-Droid because the source author signature is the same.
